core/src/main/java/de/danoeh/antennapod/core/storage/DBUpgrader.java

```diff
@@ -268,6 +267,26 @@ static void upgrade(final SQLiteDatabase db, final int oldVersion, final int new
             db.execSQL("ALTER TABLE " + PodDBAdapter.TABLE_NAME_FEEDS
                     + " ADD COLUMN " + PodDBAdapter.KEY_CUSTOM_TITLE + " TEXT");
         }
+        if (oldVersion < 1060596) {
+            db.execSQL("ALTER TABLE " + PodDBAdapter.TABLE_NAME_FEEDS
+                    + " ADD COLUMN " + PodDBAdapter.KEY_IMAGE_URL + " TEXT");
+            db.execSQL("ALTER TABLE " + PodDBAdapter.TABLE_NAME_FEED_ITEMS
+                    + " ADD COLUMN " + PodDBAdapter.KEY_IMAGE_URL + " TEXT");
+
+            db.execSQL("UPDATE " + PodDBAdapter.TABLE_NAME_FEED_ITEMS + " SET " + PodDBAdapter.KEY_IMAGE_URL + "  = ("
+                    + " SELECT " + PodDBAdapter.KEY_DOWNLOAD_URL
+                    + " FROM " + PodDBAdapter.TABLE_NAME_FEED_IMAGES
+                    + " WHERE " + PodDBAdapter.TABLE_NAME_FEED_IMAGES + "." + PodDBAdapter.KEY_ID
+                    + " = " + PodDBAdapter.TABLE_NAME_FEED_ITEMS + "." + PodDBAdapter.KEY_IMAGE + ")");
+
+            db.execSQL("UPDATE " + PodDBAdapter.TABLE_NAME_FEEDS + " SET " + PodDBAdapter.KEY_IMAGE_URL + " = ("
+                    + " SELECT " + PodDBAdapter.KEY_DOWNLOAD_URL
+                    + " FROM " + PodDBAdapter.TABLE_NAME_FEED_IMAGES
+                    + " WHERE " + PodDBAdapter.TABLE_NAME_FEED_IMAGES + "." + PodDBAdapter.KEY_ID
+                    + " = " + PodDBAdapter.TABLE_NAME_FEEDS + "." + PodDBAdapter.KEY_IMAGE + ")");
+
+            db.execSQL("DROP TABLE " + PodDBAdapter.TABLE_NAME_FEED_IMAGES);
```

Comment:
Do we intend to shrink the database file with the upgrade?

Using sqlite3 command line on windows, I noticed the db file is shrinked after `DROP TABLE` is done, as it `VACCUM` has been done. But I am unclear if the same behavior applies to Android environment.

Reply:
`VACUUM` is a quite dangerous operation. Mfietz and I decided that we do not want to execute that currently. In my case, the database shrinked significantly even without `VACUUM`

Reply:
Good to hear - my main concern is the issue should involve a deliberate decision, which is the case here.